### ISSUE TITLE
Nothing on screen

### ISSUE THREAD

**Issue by jtiscione** [Harbys/pico-ssd1306#1]
Compared to the other C++ libraries out there, I've had better luck integrating this library into an external project and getting it to compile, but I'm not seeing anything on the screen.

This is the 128x32 OLED I'm using: https://www.amazon.com/DSD-TECH-OLED-Display-Arduino/dp/B07D9H83R4

SDA pin is GPIO 4, SCL pin is GPIO 5.

```
#include "pico-ssd1306/ssd1306.h"
#include "hardware/i2c.h"

#define I2C_PORT i2c0
#define I2C_PIN_SDA 4
#define I2C_PIN_SCL 5

int main() {
  i2c_init(I2C_PORT, 1000000); //Use i2c port with baud rate of 1Mhz
  gpio_set_function(I2C_PIN_SDA, GPIO_FUNC_I2C);
  gpio_set_function(I2C_PIN_SCL, GPIO_FUNC_I2C);
  gpio_pull_up(I2C_PIN_SDA);
  gpio_pull_up(I2C_PIN_SCL);
  sleep_ms(250); // Allow time for initialization
  // Create a new display object
  pico_ssd1306::SSD1306 display = pico_ssd1306::SSD1306(I2C_PORT, 0x3D, pico_ssd1306::Size::W128xH32);
  for (int y=0; y < 32; y++) {
    display.setPixel(20, y);
  }
  display.sendBuffer();
  sleep(10000);
}
```
Compiles and runs beautifully but the screen is blank. Am I missing something here?

I know they're wired up correctly to the I2C pins physically, because I can get the martinkooij driver's example code to run and display stuff on screen. I've tried I2C address 0x3C and 0x3D.

**Comment by Harbys** [comment]
1. You're missing pico/stdlib.h import, which is required to use gpio_set_function, sleep_ms etc...
2. sleep function doesn't exist, use sleep_ms
3. Address for 128x32 displays is usually 0x3C, this one works for me

changing the code to 
```
#include "pico-ssd1306/ssd1306.h"
#include "hardware/i2c.h"
#include "pico/stdlib.h"

#define I2C_PORT i2c0
#define I2C_PIN_SDA 4
#define I2C_PIN_SCL 5

int main() {
    i2c_init(I2C_PORT, 1000000); //Use i2c port with baud rate of 1Mhz
    gpio_set_function(I2C_PIN_SDA, GPIO_FUNC_I2C);
    gpio_set_function(I2C_PIN_SCL, GPIO_FUNC_I2C);
    gpio_pull_up(I2C_PIN_SDA);
    gpio_pull_up(I2C_PIN_SCL);
    sleep_ms(250); // Allow time for initialization
    // Create a new display object
    pico_ssd1306::SSD1306 display = pico_ssd1306::SSD1306(I2C_PORT, 0x3C, pico_ssd1306::Size::W128xH32);
    for (int y=0; y < 32; y++) {
        display.setPixel(20, y);
    }
    display.sendBuffer();
    sleep_ms(10000);
}
```
draws a line for me. Please also recheck connections as gpio pins 4, 5 are physical pins 6, 7

If you will continue to have issues let me know and I can send you my breadboard photos etc

**Comment by jtiscione** [comment]
I set up a second Pico with a simpler, similar circuit and the code here (plus the initial stdio_init_all() invocation), and it worked fine. The program that I'm trying to integrate this into is more complex- it's processing ADC on one core and sending junk to an LED strip through PIO on the other. (And it did include the stdlib.h include and the call to stdio_init_all() BTW- those apparently didn't make it into my cut and paste hack job above.)
It was still giving me trouble in this larger program. I checked every #include and #define ten times.
Suddenly I got it to work- I'm not sure how, but now I can't reproduce the blank screen and I can reliably draw lines, and everything else still works. I think I may have perturbed something in CMakeLists.txt.
